scripts/tasks/merge.js

```diff
@@ -1,7 +1,7 @@
 module.exports = function merge(obj1, obj2) {
   const merged = Object.assign({}, obj1);
 
-  for (prop in obj2) {
+  for (let prop in obj2) {
```

Comment:
`prop` isn't reassigned in this loop
```suggestion
  for (const prop in obj2) {
```

Reply:
**Note:** lot of this code is very old (style, logic) meaning I didnt wrote it. The scope of this PR was to enable type checking and proper domain boundaries.